File: 10056 What is the Probability ?/solution.java
import java.io.*;

public class Main {
	public static void main(String[] args) throws IOException {
	    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StreamTokenizer st = new StreamTokenizer(br);
        st.nextToken();
        int S = (int)st.nval;
        StringBuilder output = new StringBuilder();
        while(S-- > 0) {
            st.nextToken();
            int N = (int)st.nval;
            st.nextToken();
            double p = st.nval;
            st.nextToken();
            int I = (int)st.nval;
            output.append(String.format("%.4f\n", p > 0 ? p * Math.pow(1 - p, I - 1) / (1 - Math.pow(1 - p, N)) : p));
        }
        System.out.print(output);
	}
}

File: 10056 What is the Probability ?/solution.c
#include <stdio.h>
#include <math.h>

int main() {
	int S;
	scanf("%d", &S);
	while(S--) {
		int N, I;
		double p;
		scanf("%d %lf %d", &N, &p, &I);
		printf("%.4lf\n", p > 0 ? p * pow(1 - p, I - 1) / (1 - pow(1 - p, N)) : p);
	}
	return 0;
}
